config/knex/migrations/20180626083318_create_session_table.js

```diff
@@ -0,0 +1,20 @@
+
+exports.up = function(knex, Promise) {
+    return knex.schema.createTable('session', function(table) {
+        table.increments('id').primary()
+        table.integer('machine_id').unsigned().references('id').inTable('machine')
+        .onUpdate('CASCADE')
+        .onDelete('CASCADE')
+        table.integer('user_id').unsigned().references('id').inTable('user')
+        .onUpdate('CASCADE')
+        .onDelete('CASCADE')
+        table.integer('pid').notNullable()
+        table.dateTime('start_time').notNullable()
+        table.dateTime('end_time').notNullable()

```

Comment:
Similarly, can we make this one, `end_time', allow null entries?

I think it'll make sense to create a record for an active user-process, and then just update its `end_time' later on once we confirm the process is no longer around. 

That'll also give us an easy means of querying the current processes for a machine/room since we can just search for all processes where `end_time' is still null.